CONTRIBUTING.md

```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
+
+1. Prefer the use of const pointer unless the pointer is to a value that is being modified
+
+1. The `restrict` keyword is not allowed
+
+### 7. Initialization
+
+1. An automatic variable (e.g. a local variable) must not be read before it has been initialized
+
+1. Enclose initializations for array, structure or union variables with values enclosed in braces:
+	1. `int16_t my_array[ 4 ] = { 0, 1, 2, 3 };`
+	1. `char` Arrays may also be initialized with a string literal
+	1. The stipulation is that the initializer must be internally consistent with the target array, structure or union
+	1. If an array is statically initialized the size must be explicitly stated as in `my_array[ 4 ]` above
+	1. Arrays shall not be partially initialized
+
+1. Static initialization shall not include any side effects (e.g. pointer dereference and increment which is allowed at run-time)
+
+### 8. Pointers
+
+1. Pointer conversions are allowed when:
+	1. The conversion is between a pointer and bool
+	1. The conversion is from a void type and an intrinsic type
+	1. The conversion is between compatible types
+	1. The conversion is to a char or unsigned char so that the bytes of an object can be accessed directly
+
+1. Function pointers cannot be converted
+
+1. Pointer conversions between typed objects and integers are not allowed
+
+1. Problems with pointer conversions are mostly concerned with alignment problems and accessing beyond the bounds of the size of the object
+
+1. Always use the macro NULL to reference an invalid or uninitialized pointer
+
+### 9. Expressions
+
+1. Always use parentheses to enforce explicit precedence
+
+1. For shift operations, the amount shifted must be &gt;= 0 and &lt; sizeof( value being shifted )
+
+1. Do not use the comma operator
+	1. Multiple initialization on the same line
+	1. For loops must iterate and act upon a single variable
+
+1. Always check for overflow on arithmetic operations on unsigned variables
+
+### 10. Side Effects
+
+1. Initializer lists must not have side effects
+	1. No modification of pointers during initialization
+	1. Make all statements explicit
+
+1. Check macros to prevent side-effects in calls (e.g. What happens when you call `DO_COPY_INDEX ( i++ )`?)
+```
+#define DO_COPY_INDEX ( x ) ( a [ ( x ) ] = b [ ( x ) ] )
+```
+
+
+1. Prefer that increment and decrement operators ( `++` and `--` ) are explicit statements instead of integral in an operation, e.g.
+	1. Macros
+	1. Logical comparisons ( `&&` and `||` )
+	1. `sizeof`
+
+1. Do not use the result of an assignment operation as a variable or input into another statement
+
+1. Logical comparisons should not contain persistent side effects such as assignments
+
+### 11. Control Statements
+
+1. Always use spaces instead of tabs
+	1. Use 4 space as the standard indent
+
+1. Include a space after an opening parenthesis, brace or bracket and a space before the closing match.
+	1. Parentheses: `( to_b || not_to_b )`

```

Comment:
I think our C-style guide has us putting parenthesis around each conditional in a multiple conditional check.  I didn't see anything about multiple-conditionals in this style-guide, though.  Is it something we want to add?